feat: add ult_ofNat

bollu opened the pull request:
- [ ] depends on: #25

tobiasgrosser commented on Std/Data/BitVec/Lemmas.lean:
```diff
@@ -117,6 +117,14 @@ private theorem lt_two_pow_of_le {x m n : Nat} (lt : x < 2 ^ m) (le : m ≤ n) :
   else
     simp [h]
 
+theorem BitVec.ult_ofNat (w : Nat) (x y : Nat)  :
```
```suggestion
theorem ult_ofNat (w : Nat) (x y : Nat)  :
```

tobiasgrosser commented:
You want 'feat: ' in the commit message.

alexkeizer commented:
I've simplified the proof by extracting two more intermediate simp-lemmas. One of which I've extracted into a separate PR, #25.

The other is a lemma about `Fin`. Maybe we bundle that one in this PR, so that it's clear why we care about a non-bitvector lemma? Or should we keep sticking to the one-lemma-per-PR workflow?

tobiasgrosser commented:
We should not pull them apart further than needed. If they are individually simple and can be reviewed in less than 5', we are good.

bollu commented:
@alexkeizer I'll PR this once your PR goes through.

alexkeizer commented:
> @alexkeizer I'll PR this once your PR goes through.

That PR was merged, and I've merged it in this branch

hargoniX committed:
refactor: erase to BEq/LawfulBEq (#661)

kim-em committed:
feat: runLinter doesn't error if nolints.json is missing (#662)

alexkeizer committed:
feat: simp-lemmas about `BitVec.cast` (#658)

Commit:
chore: lemma for BitVec.ofFin to .ofNat (#638)

alexkeizer committed:
feat: Add a simp lemma `Nat.cast x = BitVec.ofNat _ x` (#647)

Use `BitVec.ofNat` as the simp-normal form of both ways   to convert a `Nat` into a `BitVec` (`OfNat` and now also `Nat.cast`).

alexkeizer committed:
feat: establish `allOnes` as simp normal form (#660)

Adds a lemma `negOne_eq_allOnes`, ensuring that `(-1 : BitVec w)` and `-1#w` are both (transitively) simplified to `allOnes w`.

Commit:
chore: refactor Backtrack and SolveByElim to remove Nondet dependency (#654)

This introduces an `Iterator` in `Lean.Meta` that iterates over values and updates the state when a value is retrieved.

Co-authored-by: Scott Morrison <scott.morrison@gmail.com>

bollu committed:
feat: add `ule_ofNat`, `ult_ofNat`.

bollu commented:
Opened at https://github.com/leanprover/std4/pull/664
